src/PhpSpec/Console/Command/RunCommand.php

```diff
@@ -31,7 +32,7 @@
  */
 final class RunCommand extends Command
 {
-    public function getApplication() : Application
+    public function getApplication() : ?BaseApplication
```

Comment:
This method needs to return an instance of `PhpSpec\Console\Application` instead of `Symfony\Component\Console\Application` due to needing to subsequently call `getContainer()` which is a method of the former but not a method of the latter.

Is this change necessary? I'm not saying it isn't, just curious given the impact this change makes.

Reply:
I've tried this out in [my own attempt to get Symfony 6.0 working](https://github.com/webignition/phpspec) and I can't see the need for this.

Reply:
Maybe I have to try again without these change...
But this is a different return typehint than the Symfony one and I'm not sure it can work.
Maybe just an assertion can be enough on phpspec code.

Reply:
Returning `PhpSpec\Console\Application` not only works but is necessary for the command to call `PhpSpec\Console\Application::getContainer()`.

You are correct in saying that the return type is different. An instance of `PhpSpec\Console\Application` is acceptable to return here as it is a child type of (i.e. extends) `Symfony\Component\Console\Application`. Technically speaking, this is a covariant return type and support was introduced in PHP 7.4.

Reply:
Yes you're right. But "we" have to drop support for PHP 7.3 then.

Reply:
And I think we can drop support for Symfony 3 too... It is not maintained anymore.

Reply:
@ciaranmcnulty Are you ok to drop both PHP 7.3 and Symfony 3 support?

Reply:
The PHP 7.3 concern is a valid point that I completely failed to consider and that does complicate things somewhat. But with PHP 7.3 now being EOL, I don't think it's fully unwise to consider dropping support.

Reply:
I think we should keep this returning `Application`, then do a runtime assertion in the code that it's a `BaseApplication` we got back

Reply:
I think we can keep the existing code code, cause on PHP lower than 8.0 we have not symfony 6 downloaded so this still works with no return type on symfony lower than 6.